### DISCUSSION TITLE
we need to support formdata + binary uploads!

### DISCUSSION THREAD

**Discussion by marclave** [scalar/scalar#1301] [Ideas]
we should support all these (taken from old design of our client)
<img width="579" alt="image" src="https://github.com/scalar/scalar/assets/6176314/2d51d459-cecc-436a-8b0a-22de7fd6afbd">

<img width="588" alt="image" src="https://github.com/scalar/scalar/assets/6176314/6ef7bd70-8de2-407d-8c39-67f7271171ee">
<img width="583" alt="image" src="https://github.com/scalar/scalar/assets/6176314/142f1020-350c-40e2-b681-3bb82655997b">

for binary file we can, or if a formdata has a file type we should have a button that opens the finder to add a file to, then the active state is a descriptor of the filename with an x button (@cameronrohani we can mock something up)

@hanspagel this can most likely be broken into smaller tickets :)

**Comment by edorgeville** [comment]
This is the only missing part holding back our `@fastify/swagger-ui` to `@scalar/fastify-api-reference` migration, looking forward to file uploads being supported ! 🤗

**Comment by dromzeh** [comment]
I have several endpoints that rely on formData instead of json primarily due to file uploads being involved. This would be cool to be implemented in Scalar! (fwiw, more than happy to contribute to this being implemented)

**Comment by edorgeville** [comment]
Update: #2251 implemented the UI but no file gets uploaded as of version `1.24.28`.

**Comment by hanspagel** [comment]
works already
